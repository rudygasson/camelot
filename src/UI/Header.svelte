<script>
  import HeaderScore from "./HeaderScore.svelte";

  export let Name = "";
  export let Lebenspunkte = 0;
  export let Ausdauer = 0;
  export let Astralenergie = 0;
  export let Magieresistenz = 0;

  let firstName = Name.split(" ")[0];
  let [lastName] = Name.split(" ").slice(-1);
</script>

<svelte:head>
  <title>{firstName} {lastName}</title>
</svelte:head>

<header>
  <HeaderScore icon="Lebenspunkte" score={Lebenspunkte} />
  <HeaderScore icon="Ausdauer" score={Ausdauer} />
  <div class="header__title">{firstName}</div>
  <HeaderScore icon="Astralenergie" score={Astralenergie} />
  <HeaderScore icon="Magieresistenz" score={Magieresistenz} />
</header>

<style>
  header {
    color: white;
    padding: 0.5rem;
    position: fixed;
    width: 100%;
    top: 0;
    left: 0;
    height: 3rem;
    background: #6d1919;
    display: flex;
    justify-content: space-between;
    align-items: center;
  }

  .header__title {
    margin: 0 auto;
    height: 2rem;
    font-size: 1.5rem;
  }
</style>
