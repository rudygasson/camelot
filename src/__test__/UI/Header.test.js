/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/svelte'

import Header from '../../UI/Header.svelte'

test('shows proper name and data when rendered', () => {
  const { container } = render(Header, { Name: 'Charlotte Holzer', Lebenspunkte: 30 })

  expect(container).toMatchSnapshot();
});

test('shows zeros as default header values if no values are provided', () => {
  const { getAllByRole } = render(Header);

  const levels = getAllByRole('cell');

  expect(levels.length).toBe(4);
  levels.forEach(el => expect(el.textContent).toMatch('0'));
});
