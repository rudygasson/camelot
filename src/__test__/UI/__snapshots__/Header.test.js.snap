// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows proper name and data when rendered 1`] = `
<body>
  <div>
     
    <header
      class="svelte-scd42p"
    >
      <div
        class="score svelte-1s4e1st"
        role="cell"
      >
        <svg
          aria-labelledby="Lebenspunkte"
          class="svelte-10h4f31"
          height="32"
          role="presentation"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="Lebenspunkte"
            lang="en"
          >
            Lebenspunkte icon
          </title>
          <g
            fill="currentColor"
          >
            <path
              d="M8.272,4.9526 C7.174,4.9526 6.077,5.3496 5.241,6.1446 C4.441,6.9066 4,7.9176 4,8.9916 C4,10.0646 4.441,11.0756 5.241,11.8376 L12,18.2696 L18.759,11.8376 C19.559,11.0756 20,10.0646 20,8.9916 C20,7.9186 19.559,6.9066 18.759,6.1446 C17.088,4.5546 14.368,4.5546 12.697,6.1446 L12,6.8086 L11.303,6.1446 C10.467,5.3496 9.37,4.9526 8.272,4.9526 L8.272,4.9526 Z M12,20.9996 L3.847,13.2406 C2.656,12.1076 2,10.5986 2,8.9916 C2,7.3846 2.656,5.8756 3.847,4.7416 C6.067,2.6286 9.552,2.4386 12,4.1686 C14.448,2.4386 17.933,2.6286 20.153,4.7416 C21.344,5.8756 22,7.3846 22,8.9916 C22,10.5986 21.344,12.1076 20.153,13.2406 L12,20.9996 Z"
              fill-rule="evenodd"
            />
          </g>
        </svg>
         
        <div
          class="score__value svelte-1s4e1st"
        >
          30
        </div>
      </div>
       
      <div
        class="score svelte-1s4e1st"
        role="cell"
      >
        <svg
          aria-labelledby="Ausdauer"
          class="svelte-10h4f31"
          height="32"
          role="presentation"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="Ausdauer"
            lang="en"
          >
            Ausdauer icon
          </title>
          <g
            fill="currentColor"
          >
            <rect
              height="2"
              width="2"
              x="15"
              y="19"
            />
             
            <rect
              height="2"
              width="2"
              x="15"
              y="23"
            />
             
            <path
              d="M23,11.67V4h3V2H6V4H9v7.67a2,2,0,0,0,.4,1.2L11.75,16,9.4,19.13a2,2,0,0,0-.4,1.2V28H6v2H26V28H23V20.33a2,2,0,0,0-.4-1.2L20.25,16l2.35-3.13A2,2,0,0,0,23,11.67ZM21,4v7H11V4Zm0,16.33V28H11V20.33L14.25,16,12,13h8l-2.25,3Z"
              transform="translate(0)"
            />
          </g>
        </svg>
         
        <div
          class="score__value svelte-1s4e1st"
        >
          0
        </div>
      </div>
       
      <div
        class="header__title svelte-scd42p"
      >
        Charlotte
      </div>
       
      <div
        class="score svelte-1s4e1st"
        role="cell"
      >
        <svg
          aria-labelledby="Astralenergie"
          class="svelte-10h4f31"
          height="32"
          role="presentation"
          viewBox="0 0 32 32"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="Astralenergie"
            lang="en"
          >
            Astralenergie icon
          </title>
          <g
            fill="currentColor"
          >
            <path
              d="M29.4141,24,12,6.5859a2.0476,2.0476,0,0,0-2.8281,0l-2.586,2.586a2.0021,2.0021,0,0,0,0,2.8281L23.999,29.4141a2.0024,2.0024,0,0,0,2.8281,0l2.587-2.5865a1.9993,1.9993,0,0,0,0-2.8281ZM8,10.5859,10.5859,8l5,5-2.5866,2.5869-5-5ZM25.4131,28l-11-10.999L17,14.4141l11,11Z"
            />
             
            <rect
              height="2.8284"
              transform="translate(-10.1421 7.5147) rotate(-45)"
              width="2.8284"
              x="2.5858"
              y="14.5858"
            />
             
            <rect
              height="2.8284"
              transform="translate(1.8579 12.4853) rotate(-45)"
              width="2.8284"
              x="14.5858"
              y="2.5858"
            />
             
            <rect
              height="2.8284"
              transform="translate(-1.6569 4) rotate(-45)"
              width="2.8284"
              x="2.5858"
              y="2.5858"
            />
          </g>
        </svg>
         
        <div
          class="score__value svelte-1s4e1st"
        >
          0
        </div>
      </div>
       
      <div
        class="score svelte-1s4e1st"
        role="cell"
      >
        <svg
          aria-labelledby="Magieresistenz"
          class="svelte-10h4f31"
          height="32"
          role="presentation"
          viewBox="-1 0 124 158"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title
            id="Magieresistenz"
            lang="en"
          >
            Magieresistenz icon
          </title>
          <g
            fill="currentColor"
          >
            <path
              d="m25.4 130.95c-15.904-11.42-25.399-30.02-25.399-49.77-.00040759-25.256 14.963-47.525 38.122-56.73 1.203-.478 2.426-.919 3.667-1.319l.633 2.024c-1.198.385-2.378.811-3.539 1.273-22.352 8.884-36.793 30.375-36.793 54.752-.0004 19.06 9.164 37.02 24.515 48.03l-1.206 1.74z"
            />
             
            <path
              d="m59.977 142.08c-2.692 0-5.386-.18-8.009-.54l.278-2.1c2.53.34 5.131.52 7.731.52 31.918 0 57.883-26.37 57.883-58.78 0-6.873-1.16-13.609-3.44-20.02l1.96-.724c2.37 6.645 3.57 13.624 3.57 20.744 0 33.58-26.901 60.9-59.973 60.9"
            />
             
            <polygon
              points="62.49 0 38.342 62.062 63.422 62.067 26.08 158.08 104.77 54.213 76.62 54.229 106.66 0"
            />
          </g>
        </svg>
         
        <div
          class="score__value svelte-1s4e1st"
        >
          0
        </div>
      </div>
    </header>
  </div>
</body>
`;
